be/src/agent/agent_server.cpp

```diff
@@ -67,438 +47,236 @@ AgentServer::AgentServer(ExecEnv* exec_env,
-            if (task.__isset.check_consistency_req) {
-                _check_consistency_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::UPLOAD:
-            if (task.__isset.upload_req) {
-                _upload_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::DOWNLOAD:
-            if (task.__isset.download_req) {
-                _download_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::MAKE_SNAPSHOT:
-            if (task.__isset.snapshot_req) {
-                _make_snapshot_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::RELEASE_SNAPSHOT:
-            if (task.__isset.release_snapshot_req) {
-                _release_snapshot_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::MOVE:
-            if (task.__isset.move_dir_req) {
-                _move_dir_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::RECOVER_TABLET:
-            if (task.__isset.recover_tablet_req) {
-                _recover_tablet_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
-            }
-            break;
-        case TTaskType::UPDATE_TABLET_META_INFO:
-            if (task.__isset.update_tablet_meta_info_req) {
-                _update_tablet_meta_info_workers->submit_task(task);
-            } else {
-                status_code = TStatusCode::ANALYSIS_ERROR;
+                ret_st = Status::InvalidArgument(strings::Substitute(
+                        "task(signature=$0) has wrong request member", signature));
             }
             break;
         default:
-            status_code = TStatusCode::ANALYSIS_ERROR;
+            ret_st = Status::InvalidArgument(strings::Substitute(
+                    "task(signature=$0, type=$1) has wrong task type", signature, task_type));
             break;
         }
+#undef HANDLE_TYPE
 
-        if (status_code == TStatusCode::ANALYSIS_ERROR) {
-            OLAP_LOG_WARNING("task anaysis_error, signature: %ld", signature);
-            error_msgs.push_back("the task signature is:" + to_string(signature) + " has wrong request.");
+        if (!ret_st.ok()) {
+            LOG(WARNING) << "fail to submit task. reason: " << ret_st.get_error_msg()
+                    << ", task: " << task;
+            break;
```

Comment:
`break` here means some of the tasks are successfully submitted, and rest of them are not?

Reply:
so If something is wrong, Didn't FE resend requests？

An ideal solution should be: each task in a batch has its own status, rather than each batch shares a status, but this is something that can be improved in the future.

Reply:
https://github.com/apache/incubator-doris/blob/a27e89065b692acd72ef2a1a5d616621923a3da3/fe/src/main/java/org/apache/doris/task/AgentBatchTask.java#L164-L168

Oh, at present, Fe does not check the return value of submit_tasks().

But there should be no problem here because this is just to check the mapping relationship between `task_type` and `req member`, which is almost unmistakable. We can wait until a separate status for each task was added in the future.

Reply:
Anyway, I removed the `break`, and add some comments and TODO.